src/test/java/com/xceptance/neodymium/util/SelenideAddonsTest.java

```diff
@@ -235,127 +301,115 @@ public void testWrapAssertionErrorWithoutMessage()
         }
     }
 
+    @Test()
+    public void testIsThrowableCausedBy()
+    {
+        Throwable causedByIOException = new RuntimeException("This is runtime exception", new AssertionError("this is assertion error", new IOException("this is final cause")));
+        Assert.assertTrue("Throwable has unexpected cause", SelenideAddons.isThrowableCausedBy(causedByIOException, IOException.class));
+    }
+
+    @Test()
+    public void testIsThrowableCausedByMessage()
+    {
+        Throwable causedByIOException = new RuntimeException("This is runtime exception", new AssertionError("this is assertion error", new IOException("this is final cause")));
+        Assert.assertTrue("Throwable has unexpected cause",
+                          SelenideAddons.isThrowableCausedBy(causedByIOException, IOException.class, "this is assertion error"));
```

Comment:
exchange `IOException.class` with `NullPointerException.class` in `SelenideAddons.isThrowableCausedBy` to make sure that the `new IOException("this is final cause")` is not rendering this test as valid

Reply:
Done